### PR TITLE
Fix last few mega branch issues.

### LINKED ISSUE #837: Issues from Mega-branch
- [x] “This event is already past” notification should be an info box that stays on the event page instead of disappearing
- [x] The controller method that updates diet doesn’t check permissions
- [x] In the Bonner Profile accordion, “Bonner Notes” should be “Add Bonner Note”, and the modal should have the “Related to Bonner” toggle on by default (when coming from that button).
- [x] The route and method names for generating a file should not still be named after sending a recommendation. Also, there is no error reporting to user if generating the file fails. The exception is caught silently, and a 200 is always returned.
- [x] The Program descriptions in the object are out of order from production. The best fix is to add url and description to the database and remove those methods from the object. The second best method is to put them back in production order.
- [x] Bonner Scholars is not plural, but is “Bonners Scholars” on the landing page

### PR DESCRIPTION
**Pull request by AndersonStettner**
Resolved all of the leftover comments from the mega branch. 
Did: 
- Changed the past event notification when a user is looking an event that has already passed.
- Added a check for the user before the user diet update.
- Changed the name of the "Bonner Notes" button to "Add Bonner Note".
- Changed how the JS checks the "Note is for Bonner".
- Made "Note is for Bonner" be checked by default if a user clicks "Add Bonner Note".
- Renamed all of the send recommendation routs so they better match their new purpose. 
- Changed the Program url and description from properties to fields.
- Created a SQL script to update the fields when it gets merged into production.
- Made sure that Bonner was not plural in "Bonner Scholars".
- Added the program an event belongs to to the header on the "Track Volunteers" page.

To test:
- Checkout the branch and reset your database
- Run the test suite and make sure all tests pass
- Run the application and make sure all the card titles match their descriptions on the landing page
- Make sure that Bonner is not plural on its card
- Make sure that if you click on the link to view the CELTS website for the programs you are taken to the correct CELTS web page
- Create an event for any of the programs, return to the landing page, find the program you created an event in, verify the button at the bottom of the card is "View Events", click the "View Events" button and verify it takes you to the correct tab on the "Events List" page
- Go to "Course Management" in the Admin section, Make sure there is a course in the "Approved" section, click the "Download File" button to verify it still works as intended, click "Send Recommendation" to verify it also still works as intended
- On "User Profile" make sure you are still able to create profile notes and that if you create a note from the Bonner section "Note for Bonner" is checked when the notes modal is opened
- Also on "User Profile" go to the "Dietary Restrictions" section and verify you are able to update the users dietary restrictions. You want to do this on a normal user (ayisie) and you want to change to an admin (ramsayb2) and a student staff user (neillz) and make sure you are still able to update the basic users (ayisie) dietary restrictions
- Run the SQL script and make sure it works:
1. In the terminal type _mysql -u root -proot_ then type _use celts;_ (this makes it so you are in mysql using the celts database)
2. run the SQL script with: _source /home/{user}/celts/database/program-url-and-description.sql_
- Verify the program on the "Track Volunteers" page is the correct program

### CODE REVIEW

**Comment by AndersonStettner** [comment]
[Issues from Mega-branch #837](https://trello.com/c/ZGVSHUPX/598-issues-from-mega-branch-837)

File: app/templates/admin/createEvent.html [hunk 1]
```diff
@@ -44,7 +44,7 @@ <h3>CELTS-Sponsored Event</h3>
       {% endif %}
     <h1 id="title">{{page_title}}</h1>
     {% if isPastEvent %}
-      <div class="alert alert-danger" role="alert">This event is in the past.</div>
+      <h3>This is a past event!</h3>
```
Comment (bledsoef) [inline app/templates/admin/createEvent.html]: No, bad. This is also not a change that is reflected on trackVolunteers or viewEvent.

**Review by bledsoef — CHANGES_REQUESTED** [review]

**Comment by qasema** [comment]
On the landing page, when clicking on "learn more" of any card. It should take you to the corresponding celts website page for that program. Now, it just redirects to the same landing page.

**Comment by qasema** [comment]
fix "Bonner Scholar" to "Bonner Scholars" Scholar should be plural

**Comment by qasema** [comment]
still fix "Bonner Scholar" to "Bonner Scholars" Scholar should be plural

**Review by qasema — DISMISSED** [review]

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**